breadbox/breadbox/crud/dataset.py

```diff
@@ -1367,37 +1374,31 @@ def delete_dataset(
     return True
 
 
-def get_features(
-    db: SessionWithUser, user: str, dataset_ids: List[str], feature_ids: List[str]
-) -> list[DatasetFeature]:
```

Comment:
This crud function needed some refactoring to work with dataset given IDs, and I updated the contract to be more generalizable while I was at it (`get_dataset_feature_by_given_id`). This will also be helpful for supporting slice queries in breadbox.